extensions/warp-ipfs/examples/messenger.rs

```diff
@@ -1026,6 +1016,66 @@ enum Command {
+        Command::Attach(vec![]),
+        Command::Download(Uuid::nil(), "".into(), PathBuf::default()),
+        Command::React(Uuid::nil(), ReactionState::Add, "".into()),
+        Command::Status(Uuid::nil()),
+        Command::Pin(PinTarget::All),
+        Command::Unpin(PinTarget::All),
+        Command::CountMessages,
+    ];
+    let mut help = String::from("List of all commands:\n");
+    for cmd in all_commands.iter() {
+        let cmd_help = match cmd {
+        Command::CreateConversation(_) => "/create <did> - create conversation with another user",
+        Command::AddRecipient(_) => "/add-recipient <did> - add recipient to conversation",
+        Command::RemoveRecipient(_) => "/remove-recipient <did> - remove recipient from conversation",
+        Command::CreateGroupConversation(_, _) => "/create-group <name> <did> ... - create group conversation with other users",
+        Command::RemoveConversation(_) => "/remove-conversation - delete current conversation. This will delete it on both ends",
+        Command::SetConversation(_) => "/set-conversation <id> - switch to a conversation",
+        Command::SetConversationName(_) => "/set-conversation-name <name> - set conversation name",
+        Command::ListConversations => "/list-conversations - list all active conversations",
+        Command::ListReferences(_) => "/list-references <lower-range> <upper-range> - list all messages in the conversation. This provides more info",
+        Command::ListMessages(_) => "/list <lower-range> <upper-range> - list all messages in the conversation. This provides more info",
+        Command::ListPages(_) => "/list-pages <lower-range> <upper-range> - list all messages in the conversation. This provides more info",
+        Command::RemoveMessage(_) => "/remove-message <message-id> - remove message from conversation",
+        Command::GetFirst => "/get-first - get first message in conversation",
+        Command::GetLast => "/get-last - get last message in conversation",
+        Command::Search(_) => "/search <keywords> - search for messages in conversation",
+        Command::EditMessage(_, _) => "/edit <message-id> <message> - edit message in the conversation",
+        Command::Attach(_) => "/attach <paths> - attach files to conversation",
+        Command::Download(_, _, _) => "/download <message-id> <file> <path> - download file from conversation",
+        Command::React(_, _, _) => "/react <add | remove> <message-id> <emoji> - add or remove reaction to a message",
+        Command::Status(_) => "/status <message-id> - get message status",
+        Command::Pin(_) => "/pin <all | message-id> - pin a message in a the conversation.",
+        Command::Unpin(_) => "/unpin <all | message-id> - unpin a message in the conversation.",
+        Command::CountMessages => "/count-messages - count messages in the conversation",
+    };
+        help.push_str(cmd_help);
+        help.push('\n');
+    }
+
+    help
+}
+
```

Comment:
I wonder if we can have `Command` implement `Display` in some way (maybe via `derive_more` crate) and then create the results by converting `cmd` to a string. That way, we can have the command have an associated help for it. 

We could probably even go as far as making a macro that would build this out so we can enforce all the commands apart of `Command`.

Reply:
> I wonder if we can have `Command` implement `Display` in some way (maybe via `derive_more` crate) and then create the results by converting `cmd` to a string.

I thought of that but 2 things:

1. We want to display help for all commands, so we'll need to loop through anyway and it won't save us much on LOC (if at all).
2. The strings here are not really string form of the command variants so it'll be strange to use Display/ToString impl for this IMO.

> We could probably even go as far as making a macro that would build this out so we can enforce all the commands apart of `Command`.

Sure, would be great to automate this but maybe another day. I want to get to function changes fast. I've spent enough time cleaning stuff. :laughing: